packages/cursorless-engine/src/util/allocateHats/getRankedTokens.ts

```diff
@@ -44,7 +49,33 @@ export function getRankedTokens(
       ),
     );
 
-    return tokens.map((token, index) => ({ token, rank: -index }));
+    return tokens;
+  });
+
+  return moveForcedHatsToFront(forcedHatMap, tokens).map((token, index) => ({
+    token,
+    rank: -index,
+  }));
```

Comment:
notice that we generate the ranks at the end now. It seems we had a bug where we were treating hats as having the same rank if they were in other editors